src/Stryker.Core/Stryker.Core/MutantFilters/ExcludeFromCodeCoverageFilter.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Stryker.Core.Mutants;
+using Stryker.Core.Options;
+using Stryker.Core.ProjectComponents;
+
+namespace Stryker.Core.MutantFilters
+{
+    public class ExcludeFromCodeCoverageFilter : IMutantFilter
+    {
+        IEnumerable<Mutant> IMutantFilter.FilterMutants(IEnumerable<Mutant> mutants, FileLeaf file,
+            StrykerOptions options) =>
+            mutants.Where(m => !Exclude(m.Mutation.OriginalNode));
+
+        private static bool Exclude(SyntaxNode node) => 
+            HasExcludeFromCodeCoverageAttribute(node) || node.Parent != null && Exclude(node.Parent);
+
+        private static bool HasExcludeFromCodeCoverageAttribute(SyntaxNode node)
+        {
+            switch (node)
+            {
+                case MemberDeclarationSyntax m:
+                    return HasExcludeFromCodeCoverageAttribute(m);
+                default:
+                    return false;
+            }
+        }
+
+        private static bool HasExcludeFromCodeCoverageAttribute(MemberDeclarationSyntax m) =>
+            m.AttributeLists.Any(attr =>
+                attr.Attributes.Any(a => Regex.IsMatch(a.ToFullString(), @"^(?:System\.Diagnostics\.CodeAnalysis\.)?ExcludeFromCodeCoverage(?:Attribute)?$")));
```

Comment:
Not sure if there's a better way to validate attributes without to much context.

Reply:
I don't think there is. We won't calculate the semantic model for each file because of the performance costs. So we have no way of getting the full name of the attribute for now.

Reply:
By that I changed it to `ToString`